### DISCUSSION TITLE
Unable to find appropriate mortar segment during linear search!

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#18724] [Q&A Modules: General] [answered]
I have developed a contact model, and I get the following error when I run the code:

Unable to find appropriate mortar segment during linear search!

Is this type of error due to the distance of the colliding parts from each other, or maybe a time stepping issue?

**Comment by GiudGiud** [comment]
@jbadger95 @lindsayad

**Comment by jbadger95** [comment]
I'm going to need more info about your setup, can you provide your input file?

**Reply by TLWise** [reply]
Yes. Thank you for taking a look.
[ParticleCladdClash.e.txt](https://github.com/idaholab/moose/files/7087757/ParticleCladdClash.e.txt)
[ParticleCladdClash.i.txt](https://github.com/idaholab/moose/files/7087758/ParticleCladdClash.i.txt)

**Comment by jbadger95** [comment, marked as answer]
Sorry for the delay, I've been tied up with some other things but I have an answer! The problem is that you have nodes on your primary surface that project to exactly the same place on the secondary surface (mortar subdivides secondary elements where primary elements project so it's trying to subdivide twice at the same point). The problem is where your bottom and sides meet. To fix this you can change your geometry, I'd probably extend the side rectangles to the bottom of the bottom rectangle then merge the all surfaces before you generate your mesh. Alternatively you could leave a small gap between the bottom and sides so that the nodes at the corners don't coincide.

We can probably address this case in the source to avoid problems in the future but for now your best bet is just to modify geometry  so there are no redundant nodes.
